### PR TITLE
Continue adding Bazel support/5

### PR DESCRIPTION
**Pull request by ffoulkes**
- Added Bazel support for ES2K.
- Updated switchapi/es2k to reflect header file changes.
- Defined a target (`//:krnlmon`) to build the Kernel Monitor.
- Replaced filegroups with libraries that aggregate the object files for a package.
- Removed or disabled unused variables and functions.

----

It is now possible to run the unit tests for either target:

```bash
# export SDE_INSTALL=<path to es2k sde>
bazel test --define target=es2k //switchlink:all //switchsde:all
```
```
# export SDE_INSTALL=<path to dpdk sde>
bazel test --define target=dpdk //switchlink:all //switchsde:all
```
The Kernel Monitor build is still under development.

### CODE REVIEW

File: bazel/es2k.bzl
Comment (5abeel) [inline bazel/es2k.bzl:2]: Fix date here

File: bazel/external/es2k.BUILD
Comment (5abeel) [inline bazel/external/es2k.BUILD:3]: Fix date year

**Review by 5abeel — CHANGES_REQUESTED** [review]
Minor change req: Edited files should have updated dates 2024

File: bazel/es2k.bzl
Comment (ffoulkes) [reply bazel/es2k.bzl:2]: The file was lifted verbatim from the Stratum repository. It's not new IP.

File: bazel/external/es2k.BUILD
Comment (ffoulkes) [reply bazel/external/es2k.BUILD:3]: File copied verbatim from Stratum repository.

**Comment by ffoulkes** [comment]
@5abeel 

1. As a rule, one does not block a commit over a copyright date. (I assume that's why you blocked it. If not, it would be appropriate to provide an explanation.
2. I did not update the copyright dates because the files had been copied over from the Stratum repository, without modification.
3. I have since modified one of the files, and I changed the copyright date at that time.
4. This is a Draft PR. I'm still working on it.

**Comment by 5abeel** [comment]
> @5abeel
> 
> 1. As a rule, one does not block a commit over a copyright date. (I assume that's why you blocked it. If not, it would be appropriate to provide an explanation).
> 2. I did not update the copyright dates because the files were copied without modification from the Stratum project.
> 3. I have since modified one of the files, and I changed the copyright date at that time.
> 4. This is a Draft PR. I'm still working on it.

Didnt mean to block the PR - that was an error.
Will review and approve once this PR is moved from draft to ready-to-review.

**Review by 5abeel — APPROVED** [review]
LGTM

**Pull request merged**
